change-streams/reacting-to-database-changes-with-mongodb-change-streams-and-go.md

```diff
@@ -0,0 +1,204 @@
+	if err != nil {
+		panic(err)
+	}
+	waitGroup.Add(1)
+	routineCtx, cancelFn := context.WithCancel(context.Background())
+	go iterateChangeStream(routineCtx, waitGroup, episodesStream)
+
+	waitGroup.Wait()
+}
+```
+
+A few things are happening in the above code. We've moved the stream iteration into a separate function to be used in a goroutine. However, running the application would result in it terminating quite quickly because the `main` function will terminate not too longer after creating the goroutine. To resolve this, we are making use of a `WaitGroup`. In our example, the `main` function will wait until the `WaitGroup` is empty and the `WaitGroup` only becomes empty when the goroutine terminates.
+
+Making use of the `WaitGroup` isn't an absolute requirement as there are other ways to keep the application running while watching for changes. However, given the simplicity of this example, it made sense in order to see any changes in the stream.
+
+To keep the `iterateChangeStream` function from running indefinitely, we are creating and passing a context that can be canceled. While we don't demonstrate canceling the function, at least we know it can be done.
+
+## Complicating the Change Stream with the Aggregation Pipeline
+
+In the previous example, the aggregation pipeline that we used was as basic as you can get. In other words, we were looking for any and all changes that were happening to our particular collection. While this might be good in a lot of scenarios, you'll probably get more out of using a better defined aggregation pipeline.
+
+Take the following for example:
+
+```go
+matchPipeline := bson.D{
+    {
+        "$match", bson.D{
+            {"operationType", "insert"},
+            {"fullDocument.duration", bson.D{
+                {"$gt", 30},
+            }},
+        },
+    },
+}
+
+episodesStream, err := episodesCollection.Watch(context.TODO(), mongo.Pipeline{matchPipeline})
+```
+
+In the above example, we're still watching for changes to the `episodes` collection. However, this time we're only watching for new documents that have a `duration` field greater than 30. Any other insert or other change stream operation won't be detected.
+
+The results of the above code, when a match is found, might look like the following:
+
+```
+map[_id:map[_data:825E4F03CF000000012B022C0100296E5A1004D960EAE47DBE4DC8AC61034AE145240146645F696400645E4F03A01C9D44000063CCBD0004] clusterTime:{1582236623 1} documentKey:map[_id:ObjectID("5e4f03a01c9d
+44000063ccbd")] fullDocument:map[_id:ObjectID("5e4f03a01c9d44000063ccbd") description:a quick start into mongodb duration:35 podcast:1234 title:getting started with mongodb] ns:map[coll:episodes db:qui
+ckstart] operationType:insert]
+```
+
+With change streams, you'll have access to the MongoDB aggregation pipeline and its operators. You can learn more about what's available in the [official documentation](http://docs.mongodb.com/manual/changeStreams/#modify-change-stream-output).
```

Comment:
The part "you'll have access to the MongoDB aggreation pipeline and its operators" still suggests to me that the full aggregation language is available. Can you change this to say "a subset of the MongoDB aggregation language"?